### PR TITLE
Add queue resource

### LINKED ISSUE #11: Implement the `queue` resource with calls to Control API SDK
┆Issue is synchronized with this [Jira Task](https://ably.atlassian.net/browse/IAC-12) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by Morganamilo**
Closes #11

### CODE REVIEW

**Comment by Morganamilo** [comment]
So all of the queue stats are going to be changing on the fly. How will terraform react to this? I don't know how to use a queue so I've not been able to test.

**Comment by kavalerov** [comment]
After looking at this I am actually wondering if we should remove the stats fields from the provider completely. My reasoning is the following.

The main purpose of the Able Terraform provider is to enable people to configure their Ably apps programmatically, using infrastructure as a code. This enables customers to replicate their configuration easily, have it under version control (which enables things like change management via change reviews), and have a consistent way of managing their infrastructure stack. Infrastructure as a code should describe the _configuration_ of the infrastructure, not the details of their internal operation.

These queue stats fields on the queues object serve a completely different purpose and use case - customers would use them to observe the operation of the queue. They are read only - even if you look at Control API reference, only the following is included in the payload for creating or updating the queue:

```
{
  "name": "My queue",
  "ttl": 60,
  "maxLength": 10000,
  "region": "us-east-1-a"
}
```

Therefore **I suggest we expose only these writable fields in the queues resource, and do not expose read only and dynamically changing ones.**

**Comment by Morganamilo** [comment]
Sounds good. Obviously not all read only fields, as ID and stomp_destination have value :P

But as some one who doesn't know too much terraform stuff this sounds valid.

**Comment by kavalerov** [comment]
Yeah, I guess I meant more "dynamically changing read only fields", or something like that - i.e. ID shouldn't really change throughout operation of the queue, as well as stomp_destination I think.

File: internal/provider/resource_ably_queue_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+				Config: testAccAblyQueueConfig(app_name, ably_control_go.NewQueue{
+					Name:      queue_name + "new",
+					Ttl:       99,
+					MaxLength: 83,
+					Region:    ably_control_go.UsEast1A,
+				}),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("ably_app.app0", "name", app_name),
+					resource.TestCheckResourceAttr("ably_queue.queue0", "name", queue_name+"new"),
+					resource.TestCheckResourceAttr("ably_queue.queue0", "ttl", "99"),
```
Comment (graham-russell) [inline internal/provider/resource_ably_queue_test.go:44]: Ran the Acceptance test. Got an error stating that TTL should be => 60. 
`
=== RUN   TestAccAblyQueue
    resource_ably_queue_test.go:15: Step 2/2 error: Error running apply: exit status 1
        
        Error: Error creating Resource
        
          with ably_queue.queue0,
          on terraform_plugin_test.tf line 19, in resource "ably_queue" "queue0":
          19: resource "ably_queue" "queue0" {
        
        Could not create resource, unexpected error: /apps/IPGsPg/queues: Invalid
        resource: code 40022: status code: 422 see: https://help.ably.io/error/40022
          ttl:
            must be less than or equal to 60
`

Comment (graham-russell) [inline internal/provider/resource_ably_queue_test.go:44]:
```suggestion
					resource.TestCheckResourceAttr("ably_queue.queue0", "ttl", "30"),
```

Comment (Morganamilo) [reply internal/provider/resource_ably_queue_test.go:44]: I changed the number at some point, Maybe the change wasn't commited.

**Review by graham-russell — APPROVED** [review]
LGTM

**Pull request merged**